kratos/solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h

```diff
@@ -334,6 +395,135 @@ class ResidualBasedAdjointBossakScheme : public Scheme<TSparseSpace, TDenseSpace
+     * \]
+     *
+     * @param rCurrentElement           Current element
+     * @param rLHS_Contribution         Left hand side matrix (i.e. $\mathbf{\underline{K}}$)
+     * @param rRHS_Contribution         Right hand side vector (i.e. $\underline{F}$)
+     * @param rCurrentProcessInfo       Current process info
+     */
+    virtual void CalculateResidualLocalContributions(
+        Element& rCurrentElement,
+        LocalSystemMatrixType& rLHS_Contribution,
+        LocalSystemVectorType& rRHS_Contribution,
+        const ProcessInfo& rCurrentProcessInfo)
+    {
+        CalculateEntityResidualLocalContributions(
+            rCurrentElement, rLHS_Contribution, rRHS_Contribution, rCurrentProcessInfo);
+    }
+
+    /**
+     * @brief Calculates condition residual
+     *
+     * \[
+     *      \underline{F} = \underline{F} - \mathbf{\underline{K}}\underline{\lambda}_1
+     * \]
+     *
+     * @param rCurrentCondition         Current condition
+     * @param rLHS_Contribution         Left hand side matrix (i.e. $\mathbf{\underline{K}}$)
+     * @param rRHS_Contribution         Right hand side vector (i.e. $\underline{F}$)
+     * @param rCurrentProcessInfo       Current process info
+     */
+    virtual void CalculateResidualLocalContributions(
+        Condition& rCurrentCondition,
+        LocalSystemMatrixType& rLHS_Contribution,
+        LocalSystemVectorType& rRHS_Contribution,
+        const ProcessInfo& rCurrentProcessInfo)
+    {
+        CalculateEntityResidualLocalContributions(
+            rCurrentCondition, rLHS_Contribution, rRHS_Contribution, rCurrentProcessInfo);
+    }
+
+    /**
+     * @brief Calculate time scheme contributions from elements
+     *
+     * @param rElement
+     * @param rAdjointTimeSchemeValues1
+     * @param rAdjointTimeSchemeValues2
+     * @param rCurrentProcessInfo
+     */
+    virtual void CalculateTimeSchemeContributions(
+        Element& rElement,
+        LocalSystemVectorType& rAdjointTimeSchemeValues1,
+        LocalSystemVectorType& rAdjointTimeSchemeValues2,
+        AdjointResponseFunction& rAdjointResponseFunction,
+        const BossakConstants& rBossakConstants,
+        const ProcessInfo& rCurrentProcessInfo)
+    {
+        CalculateEntityTimeSchemeContributions(rElement, rAdjointTimeSchemeValues1,
+                                               rAdjointTimeSchemeValues2,
+                                               rCurrentProcessInfo);
+    }
+
+    /**
+     * @brief Calculates time scheme contributions from conditions
+     *
+     * @param rCondition
+     * @param rAdjointTimeSchemeValues1
+     * @param rAdjointTimeSchemeValues2
+     * @param rCurrentProcessInfo
+     */
+    virtual void CalculateTimeSchemeContributions(
+        Condition& rCondition,
+        LocalSystemVectorType& rAdjointTimeSchemeValues1,
+        LocalSystemVectorType& rAdjointTimeSchemeValues2,
+        AdjointResponseFunction& rAdjointResponseFunction,
```

Comment:
you should also change the names here: rAdjointTimeSchemeValues1 -> rAdjointTimeSchemeValues2
rAdjointTimeSchemeValues2 -> rAdjointTimeSchemeValues3

Reply:
done :)